lib/libzfsacl/zfsacltests/test_nfsv4acl.py

```diff
@@ -0,0 +1,1531 @@
+    def test_017_add_ace_file(self):
+        tfile = f'{self.MOUNTPT}/test.txt'
+        with open(tfile, 'w'):
+            pass
+        facl = libzfsacl.Acl(path=tfile)
+        orig_cnt = facl.ace_count
+        newEntry = facl.create_entry(0)
+        newEntry.entry_type = libzfsacl.ENTRY_TYPE_ALLOW
+        newEntry.who = (libzfsacl.WHOTYPE_USER, self.ZFS_ACL_STAFF1_UID)
+        newEntry.flagset = 0
+        newEntry.permset = libzfsacl.PERM_READ_DATA
+        facl.setacl(path=tfile)
+        self.assertEqual(orig_cnt + 1, facl.ace_count,
+                         "Failed to add an ACE to a file")
+
+    # Test removing an ace from a file
+    def test_018_remove_ace_file(self):
+        tfile = f'{self.MOUNTPT}/test.txt'
+        facl = libzfsacl.Acl(path=tfile)
+        orig_cnt = facl.ace_count
+        facl.delete_entry(0)
+        new_cnt = facl.ace_count
+        os.remove(tfile)
+        self.assertEqual(orig_cnt - 1, new_cnt,
+                         "Failed to delete an ACE from a file")
+
+    # Test a flag can be set on file
+    def test_019_basic_flagset(self):
+        tfile = f'{self.MOUNTPT}/test.txt'
+        with open(tfile, 'w'):
+            pass
+        facl = libzfsacl.Acl(path=tfile)
+        newEntry = facl.create_entry(0)
+        newEntry.entry_type = libzfsacl.ENTRY_TYPE_ALLOW
+        newEntry.who = (libzfsacl.WHOTYPE_USER, self.ZFS_ACL_STAFF1_UID)
+        newEntry.flagset = 0
+        newEntry.permset = libzfsacl.PERM_READ_DATA
+        facl.setacl(path=tfile)
+        facl = libzfsacl.Acl(path=tfile)
+        facl_entry0 = facl.get_entry(0)
+        os.remove(tfile)
+        self.assertEqual(facl_entry0.flagset, 0,
+                         "Failed to set basic flagset")
+
+    # Test multiple flags can be set on directory
+    def test_020_advanced_flagset(self):
+        os.makedirs(self.TDIR)
+        tdacl = libzfsacl.Acl(path=self.TDIR)
+        adv_flags = libzfsacl.FLAG_FILE_INHERIT | \
+            libzfsacl.FLAG_DIRECTORY_INHERIT | \
+            libzfsacl.FLAG_NO_PROPAGATE_INHERIT | \
+            libzfsacl.FLAG_INHERIT_ONLY
+        newEntry = tdacl.create_entry(0)
+        newEntry.entry_type = libzfsacl.ENTRY_TYPE_ALLOW
+        newEntry.who = (libzfsacl.WHOTYPE_USER, self.ZFS_ACL_STAFF1_UID)
+        newEntry.flagset = adv_flags
+        newEntry.permset = libzfsacl.PERM_READ_DATA
+        tdacl.setacl(path=self.TDIR)
+        tdacl = libzfsacl.Acl(path=self.TDIR)
+        tdacl_entry0 = tdacl.get_entry(0)
+        os.rmdir(self.TDIR)
+        self.assertEqual(tdacl_entry0.flagset, adv_flags,
+                         "FLAG_INHERITED is set by default.")
+
+    # Test no inherited ace is present by default
+    def test_021_flagset_no_inherited_ace_by_default(self):
+        os.makedirs(self.TDIR)
+        tdacl = libzfsacl.Acl(path=self.TDIR)
+        not_inherited = 0
+        for i in range(tdacl.ace_count):
+            if tdacl.get_entry(i).flagset & libzfsacl.FLAG_INHERITED == 0:
```

Comment:
Do we need brackets around `(tdacl.get_entry(i).flagset & libzfsacl.FLAG_INHERITED)`?

Reply:
Updated.